### DISCUSSION TITLE
Make helper functions to have more Db agnostic code.

### DISCUSSION THREAD

**Discussion by TommiPrami** [AliDehbansiahkarbon/EasyDBMigrator#3] [Ideas]
Like in your example could be something like this.

``` Pascal
if not TableExists('USERS') then
begin
  CreateTable('USERS').AddField('USERID', dtInt64)....
end;
```

Or how would be best to implelement and so,

Now the migration is just piece of SQL that is totally dependent on the current  Database, which is very good start tough.

**Comment by AliDehbansiahkarbon** [comment]
Good idea.
Certainly, I will add more helper functions in this regard during the time.

**Comment by AliDehbansiahkarbon** [comment]
> Like in your example could be something like this.
> 
> ```pascal
> if not TableExists('USERS') then
> begin
>   CreateTable('USERS').AddField('USERID', dtInt64)....
> end;
> ```
> 
> Or how would be best to implelement and so,
> 
> Now the migration is just piece of SQL that is totally dependent on the current Database, which is very good start tough.

Check the ORM samples.
